src/components/pageLayout/PageLayout.tsx

```diff
@@ -1,14 +1,24 @@
 import Footer from "@/components/footer/Footer";
 import Header from "@/components/header/Header";
+import { auth, signIn } from "@/auth/auth";
 
-export default function PageLayout({
+export default async function PageLayout({
   children,
 }: Readonly<{
   children: React.ReactNode;
 }>) {
+
+  const session = await auth();
+  const user = session?.user;
+
+  const handleLogin = async () => {
+    "use server";
+    await signIn();
+  };
+
   return (
     <div className="page-layout-container">
-      <Header />
+      <Header handleLogin={handleLogin} user={user} />
```

Comment:
it seems the header component simply needs to know if the user session exists or not? So instead of passing the Session user object, why not pass a boolean that indicates if there is a session or not? We could update the name of the prop.
```suggestion
      <Header handleLogin={handleLogin} isLoggedIn={!!user} />
```